crates/typst-library/src/meta/provide.rs

```diff
@@ -0,0 +1,43 @@
+use crate::prelude::*;
+
+/// Provide custom metadata for the Typst query system.
+///
+/// The Typst query system allows users to extract metadata from the document, offering both a
+/// generic selector and a specific key-value mechanism. The 'provide' command is an essential
+/// part of this mechanism, associating a single piece of metadata with the specified key.
+/// Subsequent invocations will append to a list of values for the same key.
+///
+/// Example:
+/// ```example
+/// #provide("note", "This is a note")
+///
+/// While this note has no visible output in the document, it embeds metadata into the document! This metadata can be retrieved using the 'query' command.
+/// ```
+///
+/// How to retrieve the metadata:
+/// ```sh
+/// $ typst query example.typ --key note
+/// ["This is a note"]
+/// ```
+///
+/// Display: Provide
+/// Category: meta
+#[element(Behave, Show, Locatable)]
+pub struct ProvideElem {
```

Comment:
I wonder whether `provide` is really the best name for this... how about `expose`?

Reply:
We can discuss. For me "providing information" seems to sound natural. I use the command to provide a note to someone. What's yourargument for expose? ;-)

Reply:
It exposes information to observers outside the document. People are free to observe or not observe it, expose itself doesn't care.

Reply:
Does provide care? Nobody needs to take provided information, either. As you can also access the information from within the document, <ex>pose might be misleading?

Reply:
I posted the PR on Discord. Maybe some people will express their opinion. :)

Reply:
ok

Reply:
I'm thinking maybe it should be neither. Since it is an element, perhaps it should say what it `is` and not what it `does`.

Reply:
And that would be ...

Reply:
HiddenInformation?

Reply:
More like `metadata` or something. There is already `MetaElem` but it's not exposed. I'll think some more about this.